code/game/objects/structures/holosign.dm

```diff
@@ -104,6 +104,67 @@
 	. = ..()
 	air_update_turf(TRUE)
 
+///this is a machinery to make it possible to check on the APC power to shut it down in case of an outage
+/obj/machinery/holosign/barrier/power_shield
+	name = "powered shield"
+	desc = "A shield to prevent changes of atmospheric and heat transfer"
+	icon = 'icons/effects/effects.dmi'
+	density = FALSE
+	anchored = TRUE
+	CanAtmosPass = ATMOS_PASS_NO
+	resistance_flags = FIRE_PROOF
+	///store the conductivity value of the turf is applyed so that it can be restored on removal
+	var/stored_conductivity = 0
+	///power drain from the apc, in W (so 5000 is 5 kW), per each holosign placed
+	var/power_consumption = 5000
+	var/obj/item/holosign_creator/shield_projector
+
+/obj/machinery/holosign/barrier/power_shield/Initialize(loc, source_projector)
+	. = ..()
+	if(source_projector)
+		shield_projector = source_projector
+		shield_projector.signs += src
+	air_update_turf(TRUE)
+	var/area/a = get_area(src)
+	a.addStaticPower(power_consumption, STATIC_EQUIP)
+	shield_turf()
```

Comment:
In case I wasn't clear, do check if the area is powered here, just make it scream a `stack_trace` message if so before deleting.

Reply:
ah but is not needed anymore since the check happens in the holosign projector itself and if there isnt power in the targeted area it wont even try to create the holobarrier

Reply:
in fact there isnt even a qdel in this anymore

Reply:
What if an admin spawns one?